parser/src/parser.rs

```
@@ -23,64 +26,105 @@ pub(crate) struct Parser<'a> {
 }
 
 impl<'a> Parser<'a> {
+    /// Parses input tokens into an abstract syntax tree representation.
+    pub fn parse(&mut self) -> Vec<Expr<'a>> {
```

The error is lost, keep returning a `ParseResult<Vec<Expr<'a>>` here.